parse_contributors_oo.py

```diff
@@ -0,0 +1,313 @@
+        return contrib_data
+
+    def format_url(self, url: str) -> str:
+        """Append https to the beginning of URL if it doesn't exist
+        If the url doesn't have https add it
+        If the url starts with http change it to https
+        Else do nothing
+
+        Parameters
+        ----------
+        url : str
+            String representing the url grabbed from the GH api
+
+        """
+        if not url:
+            return url  # returns empty string if url is empty
+        elif url.startswith("https://"):
+            return url
+        elif url.startswith("http://"):
+            return "https://" + url[7:]
+        else:
+            return "https://" + url
+
+    def create_new_contrib_file(self, filename: str, contrib_data: dict):
+        """Update website contrib file with the information grabbed from GitHub
+        API
+
+        Serialize contrib data from dictionary to YAML file.
+
+        Parameters
+        ----------
+
+        filename : str
+            Name of the output contributor filename ().yml format)
+        contrib_data :  dict
+            A dict containing contributor data for the website.
+
+        Returns
+        -------
+        """
+
+        with open(filename, "w") as file:
+            # Create YAML object with RoundTripDumper to keep key order intact
+            yaml = ruamel.yaml.YAML(typ="rt")
+            # Set the indent parameter to 2 for the yaml output
+            yaml.indent(mapping=4, sequence=4, offset=2)
+            yaml.dump(contrib_data, file)
+
+    def clean_yaml_file(self, filename):
+        """Open a yaml file and remove extra indent and spacing"""
+        with open(filename, "r") as f:
+            lines = f.readlines()
+
+        cleaned_lines = []
+        for line in lines:
+            if line.startswith("  "):
+                cleaned_lines.append(line[2:])
+            else:
+                cleaned_lines.append(line)
+
+        cleaned_text = "".join(cleaned_lines).replace("''", "")
+
+        with open(filename, "w") as f:
+            f.write(cleaned_text)
+
+
+"""Begin actual script """
+# Get all contribs across repos
+# Should this be an attribute in my class or is it better served as a list here?
+json_files = [
+    "https://raw.githubusercontent.com/pyOpenSci/python-package-guide/main/.all-contributorsrc",
+    "https://raw.githubusercontent.com/pyOpenSci/software-peer-review/main/.all-contributorsrc",
+    "https://raw.githubusercontent.com/pyOpenSci/pyopensci.github.io/main/.all-contributorsrc",
+]
+
+# Get contribs from website
+web_contrib_path = "https://raw.githubusercontent.com/pyOpenSci/pyopensci.github.io/main/_data/contributors.yml"
+
+
+process_contribs = processContributors(json_files, API_TOKEN)
+# Combine the cross-repo contribut data
+all_contribs_dict = process_contribs.combine_json_data()
+
+# Open the web contrib file (could also be a method)
+with urllib.request.urlopen(web_contrib_path) as f:
```

Comment:
this could be a small helper and the web contrib path could be an attributed added to the class on instantiation?

Reply:
that makes sense to me